### PR TITLE
fix: handle IPv6 host parsing in cloudstream

### LINKED ISSUE #7040: cloud/pkg/cloudstream: IPv6 host:port parsing breaks session key lookups
**What happened**:

On latest `master`, cloudstream still parses `host:port` values by splitting on `":"`, which breaks bracketed IPv6 addresses.

Current code:

- `cloud/pkg/cloudstream/tunnelserver.go`
  - `internalIP = strings.Split(r.Request.RemoteAddr, ":")[0]`
- `cloud/pkg/cloudstream/streamserver.go`
  - `sessionKey := strings.Split(r.Request.Host, ":")[0]`

For values like `[2001:db8::1]:10250`, the current result becomes `"[2001"` instead of the real IPv6 host.

This can break session key lookup and internal IP fallback for IPv6 traffic.

**What you expected to happen**:

Host and address parsing should use proper host-port parsing so IPv4 and IPv6 are both handled correctly.

**How to reproduce it (as minimally and precisely as possible)**:

1. Check out latest `master` at `555e57bb71fede21d3fc020e95f132132018986c`.
2. Run a focused repro such as:

   ```go
   fmt.Println(strings.Split("[2001:db8::1]:10250", ":")[0])
   ```

3. Actual result:

   ```text
   [2001
   ```

4. Expected result:

   ```text
   2001:db8::1
   ```

5. Same pattern exists in cloudstream request handling for:

   - `r.Request.RemoteAddr`
   - `r.Request.Host`

**Anything else we need to know?**:

I verified this on latest `master` with a focused Go repro in WSL.

This is a real source-level bug, but it may touch broader IPv6 review context because related IPv6 discussion already exists in issues like `#6699` and `#4219`.

Still, the faulty parsing here is direct and isolated:

- `tunnelserver.go`
  - fallback internal IP extraction becomes wrong for IPv6 remote addresses
- `streamserver.go`
  - session key lookup becomes wrong for IPv6 host values

**Environment**:
- Kubernetes version (use `kubectl version`):
  - Not required for this source-level repro
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`):
  - latest `master` at `555e57bb71fede21d3fc020e95f132132018986c`
- <details><summary>Cloud nodes Environment:</summary>

  - Hardware configuration (e.g. `lscpu`):
    - Not required for this repro
  - OS (e.g. `cat /etc/os-release`):
    - Not required for this repro
  - Kernel (e.g. `uname -a`):
    - Not required for this repro
  - Go version (e.g. `go version`):
    - `go version go1.26.4 linux/amd64`
  - Others:
    - Repro is source-level and based on valid IPv6 host-port input

  </details>
- <details><summary>Edge nodes Environment:</summary>

  - edgecore version (e.g. `edgecore --version`):
    - latest `master` at `555e57bb71fede21d3fc020e95f132132018986c`
  - Hardware configuration (e.g. `lscpu`):
    - Not required for this repro
  - OS (e.g. `cat /etc/os-release`):
    - Not required for this repro
  - Kernel (e.g. `uname -a`):
    - Not required for this repro
  - Go version (e.g. `go version`):
    - `go version go1.26.4 linux/amd64`
  - Others:
    - Repro is source-level and based on valid IPv6 host-port input

  </details>

### PR DESCRIPTION
**Pull request by RedZapdos123**
## Description:

Issue #7040 reported that cloudstream parsed `host:port` values by splitting on `":"`, which breaks bracketed IPv6 addresses.

This PR updates `cloud/pkg/cloudstream/tunnelserver.go` and `cloud/pkg/cloudstream/streamserver.go` to use proper host extraction for `RemoteAddr` and `Host`, so IPv4 and IPv6 addresses are both handled correctly.

It also adds regression coverage in `cloud/pkg/cloudstream/tunnelserver_test.go` and `cloud/pkg/cloudstream/streamserver_test.go` for IPv6 host parsing and IPv6 metrics session lookup.

The fix is minimal and limited to the affected cloudstream host parsing paths.

## Validation:

- `gofmt -w cloud/pkg/cloudstream/tunnelserver.go cloud/pkg/cloudstream/streamserver.go cloud/pkg/cloudstream/tunnelserver_test.go cloud/pkg/cloudstream/streamserver_test.go`
- `go test ./cloud/pkg/cloudstream -count=1`
- `go test -v ./cloud/pkg/cloudstream -run "TestGetAddressHost|TestGetMetricsDeletesConnectionOnRequestCancelWithIPv6Host" -count=1`
- `golangci-lint run ./cloud/pkg/cloudstream --timeout=10m`

Closes #7040.

## Checklist:

- [x]  I have added regression test cases for IPv6 host parsing and IPv6 metrics session lookup.
- [x] I have run validation using the focused `go test` and `golangci-lint` commands above.

## Release note:

```release-note
CloudStream now correctly parses IPv6 host:port values for session lookup and internal IP fallback.
```

### CODE REVIEW

File: cloud/pkg/cloudstream/tunnelserver_test.go [hunk 1]
```diff
@@ -305,6 +305,36 @@ func TestConnect(t *testing.T) {
...
+			name: "IPv6 host port",
+			addr: "[2001:db8::1]:12345",
+			want: "2001:db8::1",
+		},
+		{
+			name: "Host without port",
+			addr: "edge-node",
+			want: "edge-node",
+		},
+	}
```
[the remark this answers is not in the document]
Comment (RedZapdos123) [reply cloud/pkg/cloudstream/tunnelserver_test.go:344]: Added the extra edge case coverage for `getAddressHost()`:
- IPv6 without port
- bracketed IPv6 without port
- empty string

The helper already handled these cases correctly, so this update is test only and keeps the fix logic unchanged.

**Review by DoisLONG — COMMENTED** [review]
Thanks for the fix. This looks reasonable to me overall.

The bug is real: parsing `host:port` with `strings.Split(addr, ":")[0]` breaks bracketed IPv6 addresses such as `[2001:db8::1]:10250`, which can make the session key lookup and the internal IP fallback use the wrong host value.

Using `net.SplitHostPort()` in `getAddressHost()` is the right direction here, and the fallback behavior looks reasonable for addresses without ports. The current test table also covers the cases I would expect: IPv4 with port, bracketed IPv6 with port, IPv6 without port, bracketed IPv6 without port, host without port, and empty input.

The additional `getMetrics` regression test is useful as well because it verifies the actual session lookup path with an IPv6 Host value, not just the helper function.

One small metadata point before LGTM: since this is a bug fix, could you please add the normal PR metadata, such as `/kind bug` and a release-note block? After that, and with CI green, this looks good to me.

**Comment by RedZapdos123** [comment]
Thanks for the review. I have added the normal PR metadata: the kind/bug and the release note block now.

Good to know that the fix and test coverage look good.

**Review by DoisLONG — COMMENTED** [review]
/lgtm
